answerking_app/tests/test_items.py

```diff
@@ -198,20 +216,56 @@ def test_post_invalid_price_returns_bad_request(self):
         actual = response.json()
 
         # Assert
-        self.assertEqual(expected_failure_error, actual)
+        self.assertIsInstance(actual.pop("traceId"), str)
+        self.assertEqual(expected, actual)
         self.assertEqual(response.status_code, 400)
 
     def test_post_invalid_description_returns_bad_request(self):
         # Arrange
-        invalid_post_data: NewItemType = {
+        invalid_post_data: ItemType = {
             "name": "Bad data",
             "price": "1.50",
             "description": "desc&",
             "stock": 100,
             "calories": 100,
         }
-        expected_failure_error: dict[str, list[str]] = {
-            "description": ["Enter a valid value."]
+        expected: DetailError = {
+            "detail": "Validation Error",
+            "errors": {"description": ["Enter a valid value."]},
+            "status": 400,
+            "title": "Invalid input.",
+            "type": "http://testserver/problems/error/",
+        }
+
+        # Act
+        response = client.post(
+            "/api/items", invalid_post_data, content_type="application/json"
+        )
+        actual = response.json()
+
+        # Assert
+        self.assertIsInstance(actual.pop("traceId"), str)
+        self.assertEqual(expected, actual)
+        self.assertEqual(response.status_code, 400)
+
+    def test_post_invalid_stock_and_price_returns_bad_request(self):
+        # Arrange
+        invalid_post_data: ItemType = {
+            "name": "Bad data",
+            "price": "1.50asd",  # type: ignore[reportGeneralTypeIssues]
```

Comment:
what happens if we put price with more than 2 decimal places? does it round or does it error? also is there a test for this?

Reply:
also does the price have to be sent as a string or can it be sent as a number?

Reply:
both strings and numbers work

Reply:
I don't want to add additional tests for the moment, because it wasn't the point of this PR, because most of the codebase will be changed by the next 2 PR and moreover because the tests needs a major refactor the code they are out of control.
I'm happy to do a refactoring to the tests with a separate ticket.